### ISSUE TITLE
question about DRQN

### ISSUE THREAD

**Issue by leao1995** [aai-institute/tianshou#584]
https://github.com/thu-ml/tianshou/blob/f13e415eb0de55baca5dc0d6fae39d6a38e8bc0b/tianshou/policy/modelfree/dqn.py#L167

It seems `state` is not used during training even when specifying a recurrent net. Am I missing something, or is it expected?

**Comment by Trinkle23897** [comment]
The state is used here:
https://github.com/thu-ml/tianshou/blob/f13e415eb0de55baca5dc0d6fae39d6a38e8bc0b/tianshou/policy/modelfree/dqn.py#L155

When training from sampled transitions of replay buffer, we first set the initial state to zero. That's why L167 doesn't use `state` as a network forward argument.

**Comment by leao1995** [comment]
yes, the `self()` function calls the `forward()` function, and we set it to default `None` during training, so it says we will always set the state to zero when training, but the `collector` is using `state` when collecting trajectories. so basically the model acts differently during training and testing. That seems not right if I understand it correctly.

**Comment by Trinkle23897** [comment]
#486

**Comment by michalgregor** [comment]
I stumbled upon this in the code myself today. This really does not seem to be correct and it appears to be implemented like that across different methods, which looks like a pretty serious problem to me. Has there been any progress on this lately (and the related issue #486)?

Also, do you guys maybe know of some benchmarks that we should test the agents on to make sure they are in fact working correctly once we make the required fixes?

**Comment by Trinkle23897** [comment]
> Has there been any progress

Unfortunately, no

> know of some benchmarks

Atari should be sufficient.
